### PR TITLE
fix-create-match

### PR DESCRIPTION
**Pull request by dogiadat**

### CODE REVIEW

File: app/models/match.rb [hunk 1]
```diff
@@ -43,39 +43,24 @@ def winner
...
-      self.errors.add :matches_of_team2,
-        I18n.t(".max_match_of_team", max_match_of_team: max_match_of_team)
-    end
-  end
-
-  def validate_match_of_two_team
-    if Match.of_team_vs_team(self.team1_id, self.team2_id, self.league_season_id).count >= 2
-      self.errors.add :match, I18n.t(".match_of_two_team")
+  def validate_match
+    if Match.of_team_be_team1(self.team1_id,self.league_season_id).of_team_be_team2(self.team2_id, self.league_season_id).any?
```
Comment (nguyenluc312) [inline app/models/match.rb]: dài quá hihi

File: app/models/match.rb [hunk 2]
```diff
@@ -43,39 +43,24 @@ def winner
...
+      self.errors.add :match, I18n.t(".match_invalid")
     end
   end
 
-  def validate_end_time
-    if self.end_time && self.end_time <= self.start_time
-      self.errors.add :end_time, I18n.t(".end_time_larger_than_start")
+  def check_start_time
+    if self.status && self.start_time
```
Comment (nguyenluc312) [inline app/models/match.rb]: you've checked status before

File: app/models/match.rb [hunk 3]
```diff
@@ -43,39 +43,24 @@ def winner
...
+      if self.not_started_yet? && self.start_time < Time.zone.now
+        self.errors.add :match, I18n.t(".start_time_must_larger_than_now")
+      elsif (self.finished? || self.is_on?) && self.start_time > Time.zone.now
```
Comment (nguyenluc312) [inline app/models/match.rb]: (self.finished? || self.is_on?) có cần thiết hơm?

File: app/views/admin/matches/_form_new.html.erb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  </div>
+  <div class="team2 col-sm-4">
+    <h2><%= t ".team 2" %></h2>
+  </div>
+</div>
+<div class="form">
+  <%= form_for [:admin, @league_season, @match] do |f| %>
+    <%= render "shared/error_messages", object: @match %>
+
+    <%= f.hidden_field :league_season_id, value: @league_season.id %>
```
Comment (nguyenluc312) [inline app/views/admin/matches/_form_new.html.erb]: really need this field?

File: app/views/admin/matches/_form_new.html.erb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      <div class="team1 form-group">
+        <%= f.select :team1_id, @teams, {include_blank: t(".chose_team1")},
+          class: "form-control" %>
+      </div>
+      <div class="team2 form-group">
+        <%= f.select :team2_id, @teams, {include_blank: t(".chose_team2")}, class: "form-control" %>
+      </div>
+    </div>
+    <div class="form-group start-time">
+      <label><%= t ".start_time" %></label>
```
Comment (nguyenluc312) [inline app/views/admin/matches/_form_new.html.erb]: user f.label. using <label> is not synchronous with our code

File: app/assets/javascripts/new_matches.coffee [hunk 6]
```diff
@@ -0,0 +1,6 @@
+$(document).on 'change', '.status', (event) ->
+  option = $('#select-status').val()
+  if  option != 'not_started_yet' && option != ''
```
Comment (luuthomvp) [inline app/assets/javascripts/new_matches.coffee]: thua dau cach nay

File: app/controllers/admin/matches_controller.rb [hunk 7]
```diff
@@ -1,7 +1,7 @@
 class Admin::MatchesController < ApplicationController
   load_and_authorize_resource
 
-  before_action :load_league_season, :load_teams, except: [:index, :destroy]
+  before_action :load_league_season, :load_teams, :load_statuses, except: [:index, :destroy]
```
Comment (huongnguyenmta) [inline app/controllers/admin/matches_controller.rb]: over 80 letter

File: app/views/admin/matches/_form_new.html.erb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+      <div class="team2-goals">
+        <%= f.number_field :team1_goal, step: 1, min: 0,
+          value: @match.team1_goal || 0, class: "form-control",
+          placeholder: t(".team1") %>
+      </div>
+    </div>
+    <div class="clearfix"></div>
+
+    <div class="form-group odds">
+      <label>Odds</label>
```
Comment (huongnguyenmta) [inline app/views/admin/matches/_form_new.html.erb]: I18n

File: app/views/admin/matches/_form_new.html.erb (hunk 4 above)
Comment (dogiadat) [inline app/views/admin/matches/_form_new.html.erb]: yes

File: app/controllers/admin/matches_controller.rb [hunk 9]
```diff
@@ -22,29 +26,33 @@ def edit
...
       flash[:success] = t ".success"
-      redirect_to @match
+      redirect_to [:edit, :admin, @match]
     else
       render :edit
     end
   end
 
+  def show
+
```
Comment (luuthomvp) [inline app/controllers/admin/matches_controller.rb]: remove this line

**Comment by huongnguyenmta** [comment]
OK

**Pull request merged**